docs/docs/guides/oauth2.md

```diff
@@ -0,0 +1,76 @@
+# `OAuth2`
+
+:::caution
+
+The OAuth2 feature is currently under active development.
```

Comment:
Can you add reference to the problem we've seen. Envoy GH issue etc.
And describe the problem we have

Reply:
I've done this in a section called `Upstream Issues` now.